### PR TITLE
Skirmish menu layout and map/faction selection

### PR DESCRIPTION
**Pull request by brooks42**
Got a simple skirmish screen loading, with some (incredibly trash) basic art:

image

Clicking the arrow buttons cycles between available maps or factions respectively, although the only feedback right now is the map name and index (ie "TestMap2 (1/2)") changing.

Also fixes an issue in the map custodian that stopped it from loading more than a single map.

### CODE REVIEW

**Comment by brooks42** [comment]
Marked as a draft for now since I need to build the `SKIRMISHCUSTODIAN` or whatever we want to call it to hold data for the skirmish game mode.

File: SkirmishMenuController.gd [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+func _mapSelectLeftPressed():
+	mapSelectionIndex = max(mapSelectionIndex - 1, 0) 
+	_setMapLabelText()
+	
+func _mapSelectRightPressed():
+	mapSelectionIndex = min(mapSelectionIndex + 1, MAPCUSTODIAN.masterMapArray.size() - 1) 
+	_setMapLabelText()
+
+func _player1FactionLeftPressed():
+	player1FactionIndex = max(player1FactionIndex - 1, 0)
```
Comment (JustinWinningham) [inline SkirmishMenuController.gd]: update these to allow wrap around selection?

File: project.godot [hunk 2]
```diff
@@ -20,19 +20,13 @@ _global_script_classes=[ {
 "path": "res://Scripts/Objects/Hero.gd"
 }, {
 "base": "Reference",
-"class": "Map",
-"language": "GDScript",
-"path": "res://Scripts/Objects/Map.gd"
-}, {
-"base": "Reference",
```
Comment (JustinWinningham) [inline project.godot:27]: idk why this is missing (looks automated) - but I'm guessing this is why you had that map issue

File: Globals/MAPCUSTODIAN.gd [hunk 3]
```diff
@@ -39,19 +38,28 @@ func validateAndLoadMaps():
 		#fileChecker.open(mapDataPath)
 		var errors = validateMapData(mapDataPath)
 		if errors != '0':
+			print_debug("validateMapData error" + str(errors))
 			return errors
 		errors = validateTerrainData(terrainDataPath)
+
```
Comment (JustinWinningham) [inline Globals/MAPCUSTODIAN.gd:44]: im sure there is a better way to do load error handling, but this is sufficient for now

**Review by JustinWinningham — APPROVED** [review]

**Pull request merged**